front/src/main.js

```diff
@@ -0,0 +1,423 @@
+// The Vue build version to load with the `import` command
+// (runtime-only or standalone) has been set in webpack.base.conf with an alias.
+import Vue from 'vue'
+import Vuex from 'vuex'
+import VueRouter from 'vue-router'
+import VueTimeago from 'vue-timeago'
+
+import App from './App'
+import GameList from './components/GameList'
+import TeamList from './components/TeamList'
+import MatchList from './components/MatchList'
+import TournamentList from './components/TournamentList'
+import Tournament from './components/Tournament'
+import Match from './components/Match'
+import Login from './components/Login'
+import Dashboard from './components/Dashboard'
+import Team from './components/Team'
+import auth from './auth'
+import api from './api'
+
+Vue.use(Vuex)
+Vue.use(VueRouter)
+
+Vue.use(VueTimeago, {
+  name: 'timeago',
+  locale: 'en-US',
+  locales: {
+    'en-US': require('vue-timeago/locales/en-US.json')
+  },
+})
+
+const routes = [
+  {
+    path: '/',
+    redirect: '/dashboard',
+  },
+  {
+    path: '/dashboard',
+    component: Dashboard,
+  },
+  {
+    path: '/games',
+    component: GameList,
+  },
+  {
+    path: '/teams',
+    component: TeamList,
+  },
+  {
+    path: '/matches',
+    component: MatchList,
+  },
+  {
+    path: '/tournaments',
+    component: TournamentList,
+  },
+  {
+    path: '/login',
+    component: Login,
+  },
+  {
+    path: '/logout',
+    redirect: '/login',
+  },
+  {
+    path: '/tournaments/:id',
+    name: 'tournament',
+    component: Tournament,
+  },
+  {
+    path: '/matches/:id',
+    name: 'match',
+    component: Match,
+  },
+  {
+    path: '/teams/:id',
+    name: 'team',
+    component: Team,
+  },
+]
+
+export var router = new VueRouter({
+  routes
+})
+
+function enrichMatch(match, getters) {
+  return Vue.util.extend(match, {
+    teamOne: getters.teamMap.get(match.team_one_id),
+    teamTwo: getters.teamMap.get(match.team_two_id),
+    game: getters.gameMap.get(match.game_id),
+  })
+}
+
+function enrichFriendlyMatch(match, getters) {
+  match = enrichMatch(match, getters)
+  match.owner = getters.userMap.get(match.owner_id)
+  return match
+}
+
+export var store = new Vuex.Store({
```

Comment:
Again `var`. This should be moved to a new file. Also, the parts of the state like currentUser, games, teams should be moved to new Vuex modules. http://vuex.vuejs.org/en/modules.html